kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/KieServerEnvironment.java

```diff
@@ -21,16 +21,21 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-
 public class KieServerEnvironment {
-    
+
     private static final Pattern VERSION_PAT = Pattern.compile("(\\d+)\\.(\\d+)\\.(\\d+)([\\.-].*)?");
     private static Version version;
     private static String serverId = System.getProperty(KieServerConstants.KIE_SERVER_ID);
-    private static String name = System.getProperty(KieServerConstants.KIE_SERVER_ID);
+    private static String name;
     private static String contextRoot;
 
     static {
+        if (serverId == null || serverId.length() == 0) {
+            setServerId(System.getenv("KIE_SERVER_ID"));
+        }
```

Comment:
why this needs to be in static block, wouldn't it make more sense to have it like this:

```
private static String serverId = System.getProperty(KieServerConstants.KIE_SERVER_ID, System.getenv("KIE_SERVER_ID"));
```
that way you will get the id from system properties and if not given from ENV. If you need to revert the order you could do

```
System.getenv().getOrDefault("KIE_SERVER_ID", System.getProperty(KieServerConstants.KIE_SERVER_ID);
```

Reply:
By calling setServerId, it is to avoid set null to ServerId accidentally.  (See these lines) 
https://github.com/rhtevan/droolsjbpm-integration/blob/78cb770577679cb46b1f5bbf3e38124499d3cba6/kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/KieServerEnvironment.java#L79-L84

Reply:
Committed the suggested change.